stdlib/Profile/src/Profile.jl

```diff
@@ -34,6 +34,50 @@ macro profile(ex)
     end
 end
 
+# triggers printing the report after a SIGINFO/SIGUSR1 profile request
+const PROFILE_PRINT_COND = Ref{Base.AsyncCondition}()
+function profile_printing_listener()
+    try
+        while true
+            wait(PROFILE_PRINT_COND[])
+            peek_report[]()
+        end
+    catch ex
+        if !isa(ex, InterruptException)
+            @error "Profile printing listener crashed" exception=ex,catch_backtrace()
+        end
+    end
+end
+
+# An internal function called to show the report after an information request (SIGINFO or SIGUSR1).
+function _peek_report()
+    iob = IOBuffer()
+    ioc = IOContext(IOContext(iob, stdout), :displaysize=>displaysize(stdout))
+    print(ioc, groupby = [:thread, :task])
+    Base.print(stdout, String(resize!(iob.data, iob.size)))
```

Comment:
I am not sure where you found the privacy-violating `resize!` accessor, but we have `take!` for this:
```
    Base.print(stdout, String(take!(iob.data)))
```

Reply:
Great question. I usually do `take!` I'll try to find where I copied that from.

Reply:
I can't find where I got it from.. weird. 

Applying suggestions https://github.com/JuliaLang/julia/pull/44199